lib/AST/CMakeLists.txt

```diff
@@ -99,14 +99,8 @@ add_swift_library(swiftAST STATIC
 
 # intrinsics_gen is the LLVM tablegen target that generates the include files
 # where intrinsics and attributes are declared. swiftAST depends on these
-# headers. When we build swift out of tree, this is not an issue since LLVM must
-# finish building before we build Swift. But when we are in tree, the build
-# system must be made aware of the dependency between swiftAST and
-# intrinsics_gen.
-if(NOT SWIFT_BUILT_STANDALONE)
-  get_property(CLANG_TABLEGEN_TARGETS GLOBAL PROPERTY CLANG_TABLEGEN_TARGETS)
-  add_dependencies(swiftAST intrinsics_gen ${CLANG_TABLEGEN_TARGETS})
-endif()
+# headers.
```

Comment:
Can you add a nice comment note on why this works both in and out of tree. What do you think of something like this (please word smith as you choose).

```
*NOTE* When we build swift out of tree, llvm/clang provide us with empty exported targets for ... When we build in tree we actually have a real dependency. The reason why this works is that the out of tree build guarantees that these intrinsics/tablegen targets exist and the in-tree provides the real targets.
```